src/components/notifications/NotificationConfig.tsx

```diff
@@ -149,6 +191,22 @@ const NotificationsConfig: React.FC<{ currentAccount: Account }> = ({
       )
     }
 
+    // Handle Telegram notifications
+    if (telegramNotificationConfigured) {
+      subs.notification_types = subs.notification_types.filter(
+        sub => sub.channel !== NotificationChannel.TELEGRAM
+      )
+      subs.notification_types.push({
+        channel: NotificationChannel.TELEGRAM,
+        destination: '', // Telegram destination is handled by the connection
```

Comment:
There is only one point of access to the Telegram account, that is the destination here, which you're resetting. It's set after an account is connected and not expected to be mutated.
The only thing I expect you to do here is set disabled: true , when it is disabled and disabled: false when it is enabled. No filtering out the telegram connection is needed here, so please take that out. 

Everything regarding removing and adding a telegram connection should only be managed from the reworked Connected accounts section.